src/main/java/org/wahlzeit_revisited/service/PhotoService.java

```diff
@@ -0,0 +1,122 @@
+package org.wahlzeit_revisited.service;
+
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+import org.wahlzeit_revisited.dto.PhotoDto;
+import org.wahlzeit_revisited.model.Photo;
+import org.wahlzeit_revisited.model.PhotoFactory;
+import org.wahlzeit_revisited.model.User;
+import org.wahlzeit_revisited.repository.PhotoRepository;
+import org.wahlzeit_revisited.utils.SysLog;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class PhotoService {
+
+    @Inject
+    public Transformer transformer;
+    @Inject
+    public PhotoRepository repository;
+    @Inject
+    public PhotoFactory factory;
+
+    /*
+     * business methods
+     */
+
+    public Response getPhotos() throws SQLException {
+        List<Photo> photoList = repository.findAll();
+
+        List<PhotoDto> responseDto = new ArrayList<>(photoList.size());
+        for (Photo photo : photoList) {
+            responseDto.add(transformer.transform(photo));
+        }
+        return Response.ok(responseDto).build();
+    }
+
+    public Response addPhoto(User user, byte[] photoBlob) throws SQLException {
+        Photo insertPhoto = factory.createPhoto(user.getId());
+        insertPhoto = repository.insert(insertPhoto);
+
+        // write to disk
+        String persistPath = transformer.transformToPersistPath(insertPhoto);
+        File persistFile = new File(persistPath);
```

Comment:
Allright, so we currently store the photos on the local file system as in the last version. What I am hesitant about is the fact that in the world of containers this leads to inconsistencies between file system and database pretty fast. We might want to think about it in a future PR if we better store the images in postgres.